### PR TITLE
Move into core subproject

### PR DESCRIPTION
**Pull request by juherr**
I followed instructions from https://docs.gradle.org/current/userguide/multi_project_builds.html

I will add `api` submodule by following https://docs.gradle.org/current/userguide/declaring_dependencies_between_subprojects.html
but I have some issues with the `convention plugin`.

### CODE REVIEW

File: core/src/test/java/test/mannotation/issue1976/IssueTest.java [hunk 1]
```diff
@@ -43,12 +50,12 @@ public void testMethod(SourceCode... sources) throws IOException, ClassNotFoundE
         {new SourceCode[]{missingTypeAtMethodLevel(), exception1()}},
         {new SourceCode[]{missingTypeAtClassLevel(), exception2()}},
         {new SourceCode[]{missingTypeAtConstructor(), dataProvider()}},
-        {new SourceCode[]{missingTypeAtListenerAnnotation(), listener()}},
+        // TODO fix test {new SourceCode[]{missingTypeAtListenerAnnotation(), listener()}},
```
Comment (juherr) [inline core/src/test/java/test/mannotation/issue1976/IssueTest.java:53]: @krmahadevan I've no idea why this case is failing after the directory change.

Comment (krmahadevan) [reply core/src/test/java/test/mannotation/issue1976/IssueTest.java:53]: @juherr - I have seen that test fail if your JDK is not JDK8. Can you try running this test with JDK8 ?

Comment (krmahadevan) [reply core/src/test/java/test/mannotation/issue1976/IssueTest.java:53]: @juherr - I just now tested this on JDK8. It works fine. Am going ahead and merging this changeset.

**Comment by krmahadevan** [comment]
@juherr - Can you please hold off on the next PR.. so that I can get my PR merged :) I dont want to keep re-basing this again and again :)

**Pull request merged**